File: app/src/main/java/com/gnoemes/shikimoriapp/presentation/view/search/filter/FilterResourceConverter.java
package com.gnoemes.shikimoriapp.presentation.view.search.filter;

import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem;

import java.util.List;

public interface FilterResourceConverter {

    List<FilterItem> convertFrom(List<String> actions, List<String> names, Enum[] type);
}

File: app/src/main/java/com/gnoemes/shikimoriapp/presentation/view/search/filter/FilterResourceConverter.kt
package com.gnoemes.shikimoriapp.presentation.view.search.filter

import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem

interface FilterResourceConverter {

    fun <T> convertMangaFilters(values: MutableList<String>, names: MutableList<String>, type: Array<T>): MutableList<FilterItem>

    fun <T> convertAnimeFilters(values: MutableList<String>, names: MutableList<String>, type: Array<T>): MutableList<FilterItem>
}